### PR TITLE
add Hadoop doc ToC

### PR DESCRIPTION
**Pull request by Suave**
The Hadoop doc part is too long, add ToC for better navigation.

### CODE REVIEW

File: docs/zh_cn/hadoop_java_sdk.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+  + [核心配置](#核心配置)
+  + [缓存配置](#缓存配置)
+  + [I/O 配置](#I/O-配置)
+  + [其他配置](#其他配置)
+  + [常用配置](#常用配置)
+  + [Hadoop 环境配置](#Hadoop-环境配置)
+    - [CDH6 环境配置](#CDH6-环境配置)
+    - [HDP 环境配置](#HDP-环境配置)
+  + [Flink 配置](#Flink-配置)
+* [重启相关服务](#验证)
```
Comment (xiaogaozi) [inline docs/zh_cn/hadoop_java_sdk.md:18]: Incorrect link

File: docs/zh_cn/hadoop_java_sdk.md [hunk 2]
```diff
@@ -146,7 +171,7 @@ $ make
 
 将配置参数加入到 Hadoop 配置文件 `core-site.xml` 中。
 
-#### CDH 6 环境配置
+#### CDH6 环境配置
```
Comment (xiaogaozi) [inline docs/zh_cn/hadoop_java_sdk.md:174]: Why remove space?

File: docs/zh_cn/hadoop_java_sdk.md [hunk 3]
```diff
@@ -1,5 +1,30 @@
 # 在 Hadoop 环境使用 JuiceFS Java SDK
 
+* [Haddop 兼容性](#Hadoop-兼容性)
```
Comment (xiaogaozi) [inline docs/zh_cn/hadoop_java_sdk.md:3]: How about TOC of english version doc?

**Review by xiaogaozi — CHANGES_REQUESTED** [review]

**Review by tangyoupeng — APPROVED** [review]

File: docs/zh_cn/hadoop_java_sdk.md (hunk 3 above)
Comment (Suave) [reply docs/zh_cn/hadoop_java_sdk.md:3]: added

File: docs/zh_cn/hadoop_java_sdk.md (hunk 1 above)
Comment (Suave) [reply docs/zh_cn/hadoop_java_sdk.md]: fixed

**Review by xiaogaozi — APPROVED** [review]

**Pull request merged**